orbit/cmd/orbit/orbit.go

```diff
@@ -645,6 +689,178 @@ func (d *desktopRunner) interrupt(err error) {
+//	}
+// to a map[string]string
+// this map will get compared and written to the filesystem and passed to osquery
+// this only supports simple key:value pairs and not nested structures
+func getFlagsFromJSON(flags json.RawMessage) (map[string]string, error) {
+	result := make(map[string]string)
+
+	var data map[string]interface{}
+	err := json.Unmarshal([]byte(flags), &data)
+	if err != nil {
+		log.Info().Msg(err.Error())
+		return nil, err
+	}
+
+	for k, v := range data {
+		switch t := v.(type) {
+		case string:
+			result["--"+k] = t
+		case bool:
+			result["--"+k] = strconv.FormatBool(t)
+		case float64:
+			result["--"+k] = fmt.Sprint(t)
+		}
+	}
+	return result, nil
+}
+
+// readFlagFile reads and parses the osquery.flags file on disk
+// and returns a map[string]string, of the form:
+// {"--foo":"bar","--value":"5"}
+// this only supports simple key:value pairs and not nested structures
+func readFlagFile(rootDir string) (map[string]string, error) {
+	flagfile := filepath.Join(rootDir, "osquery.flags")
+	bytes, err := ioutil.ReadFile(flagfile)
+	if err != nil {
+		return nil, fmt.Errorf("reading flagfile %s failed: %w", flagfile, err)
+	}
+	result := make(map[string]string)
+	lines := strings.Split(strings.TrimSpace(string(bytes)), "\n")
+	for _, line := range lines {
+		// skip line starting with "#" indicating that it's a comment
+		if !strings.HasPrefix(line, "#") {
+			// split each line by "="
+			str := strings.Split(strings.TrimSpace(line), "=")
+			if len(str) == 2 {
+				result[str[0]] = str[1]
+			}
+			if len(str) == 1 {
+				result[str[0]] = ""
+			}
+		}
+	}
+	return result, nil
+}
+
+// writeFlagFile writes the contents of the data map as a osquery flagfile to disk
+// given a map[string]string, of the form: {"--foo":"bar","--value":"5"}
+// it writes the contents of key=value, one line per pair to the file
+// this only supports simple key:value pairs and not nested structures
+func writeFlagFile(rootDir string, data map[string]string) error {
+	flagfile := filepath.Join(rootDir, "osquery.flags")
+	var sb strings.Builder
+	for k, v := range data {
+		if k != "" && v != "" {
+			sb.WriteString(k + "=" + v + "\n")
+		} else if v == "" {
+			sb.WriteString(k + "\n")
+		}
+	}
+	if err := ioutil.WriteFile(flagfile, []byte(sb.String()), constant.DefaultFileMode); err != nil {
+		return fmt.Errorf("writing flagfile %s failed: %w", flagfile, err)
+	}
+	return nil
+}
+
+// doFlagsUpdate, reads the flagfile on disk if it exists, and also gets flags from fleet
+// it then compare both of these flags, and returns bool indicating if the flags have been updated
+func doFlagsUpdate(orbitClient *service.Client, rootDir string, orbitNodeKey string) bool {
+	flagFileExists := true
+
+	// first off try and read osquery.flags from disk
+	osqueryFlagMapFromFile, err := readFlagFile(rootDir)
+	if err != nil && errors.Is(err, os.ErrNotExist) {
+		// flag file may not exist on disk on first "boot"
+		flagFileExists = false
+	}
```

Comment:
E.g. what if it fails to read because of permissions error, so you could do something like:
```go
if err != nil {
  if !errors.Is(err, os.ErrNotExist) {
    return false, err
  } 
  // flag file may not exist on disk on first "boot"
  flagFileExists = false
}
```